src/middlewared/middlewared/plugins/activedirectory.py

```diff
@@ -1119,9 +1119,23 @@ def check_clockskew(self, ad=None):
             self.logger.warning("Unable to find PDC emulator via DNS.")
             return {'pdc': None, 'timestamp': '0', 'clockskew': 0}
 
-        c = ntplib.NTPClient()
-        response = c.request(pdc[0]['host'])
-        ntp_time = datetime.datetime.fromtimestamp(response.tx_time)
+        try:
+            c = ntplib.NTPClient()
+            response = c.request(pdc[0]['host'])
+            ntp_time = datetime.datetime.fromtimestamp(response.tx_time)
+        except ntplib.NTPException:
+            self.logger.warning("NTP request to PDC Emulator failed. Retrying with regular DC",
+                                exc_info=True)
+
+            dc = ActiveDirectory_DNS(conf=ad, logger=self.logger).get_n_working_servers(SRV['PDC'], 1)
+            if not dc:
+                self.logger.warning("Unable to find Domain Controller via DNS.")
+                return {'pdc': None, 'timestamp': '0', 'clockskew': 0}
+
+            c = ntplib.NTPClient()
```

Comment:
Why do it the second time?

Reply:
Because I was copying and pasting while working on something else. Will fix.

Reply:
I mean why do you do NTP request the second time?